packages/shared/lib/services/connection.service.ts

```diff
@@ -849,6 +849,36 @@ class ConnectionService {
         return result.map((connection) => encryptionManager.decryptConnection(connection) as Connection);
     }
 
+    public async findConnectionsByMetadataValue({
+        metadataProperty,
+        payloadIdentifier,
+        configId,
+        environmentId
+    }: {
+        metadataProperty: string;
+        payloadIdentifier: string;
+        configId: number | undefined;
+        environmentId: number;
+    }): Promise<Connection[] | null> {
+        if (!configId) {
+            return null;
+        }
+
+        const result = await db.knex
+            .from<StoredConnection>(`_nango_connections`)
+            .select('*')
+            .where({ environment_id: environmentId, config_id: configId })
+            // escape the question mark so it doesn't try to bind it as a parameter
+            .where(db.knex.raw(`metadata->? \\? ?`, [metadataProperty, payloadIdentifier]))
```

Comment:
you can use `.whereRaw` instead of `.where(db.knex.raw...)`

Reply:
I tried this at first and I think I ran into issues with the parameters and the ? with encoding